src/service/extension-location.js

```diff
@@ -28,20 +28,44 @@ import {urls} from '../config';
  */
 let ExtensionInfoDef;
 
+/**
+ * Dynamic AMP components
+ */
+const AMP_DYNAMIC_COMPONENTS = ['amp-geo'];
+
+/**
+ * Determine whether an extension is a dynamic AMP component
+ * @param {string} extensionId
+ * @return {boolean}
+ */
+function isDynamicComponent(extensionId) {
+  return (
+    Boolean(extensionId) && AMP_DYNAMIC_COMPONENTS.indexOf(extensionId) >= 0
+  );
+}
+
 /**
  * Calculate the base url for any scripts.
  * @param {!Location} location The window's location
  * @param {boolean=} opt_isLocalDev
+ * @param {boolean=} opt_useDefaultCdn
  * @return {string}
  */
-export function calculateScriptBaseUrl(location, opt_isLocalDev) {
+export function calculateScriptBaseUrl(
+  location,
+  opt_isLocalDev,
+  opt_useDefaultCdn
+) {
   if (opt_isLocalDev) {
```

Comment:
As a future note @erwinmombay -- this is the kind of logic we should remove (local dev) when we support a development build.

Reply:
just to clarify, the development will essentially be running in its own pipeline process to separate this out correct? (which will of course clean up these logical branchings)

Reply:
Yep!